src/jit/lowerxarch.cpp

```diff
@@ -2542,47 +2561,74 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
                         {
                             ssize_t ival = op3->AsIntCon()->IconValue();
                             assert((ival >= 0) && (ival <= 255));
 
-                            if (ival <= 0x3F)
-                            {
-                                supportsAlignedSIMDLoads   = !comp->canUseVexEncoding();
-                                supportsUnalignedSIMDLoads = !supportsAlignedSIMDLoads;
-                                supportsGeneralLoads       = supportsUnalignedSIMDLoads;
-
-                                break;
-                            }
+                            supportsSIMDScalarLoads = (ival <= 0x3F);
+                            supportsGeneralLoads    = supportsSIMDScalarLoads;
                         }
-
-                        assert(supportsAlignedSIMDLoads == false);
-                        assert(supportsUnalignedSIMDLoads == false);
-                        assert(supportsGeneralLoads == false);
+                        break;
                     }
-                    else
-                    {
-                        assert(supportsAlignedSIMDLoads == false);
-                        assert(supportsUnalignedSIMDLoads == false);
 
-                        supportsSIMDScalarLoads = true;
-                        supportsGeneralLoads    = supportsSIMDScalarLoads;
-                    }
+                    // We should only get here for integral nodes.
+                    assert(varTypeIsIntegral(node->TypeGet()));
+
+                    // These intrinsics require the contained operand to be the same
+                    // size as the baseType for the containing node.
+                    //
+                    // If it isn't, we can't mark the node as contained because it
+                    // might avoid a zero or sign-extension that should have otherwise
+                    // occurred.
+                    //
+                    // We need this even though there are overloads that explicitly
+                    // take byte/sbyte/int16/uint16, since the user may be calling
+                    // the int32/uint32 overload with an explicit cast.
+                    //
+                    // Finally, SIMDScalarLoads aren't supported because these intrinsics
+                    // take an actual scalar value, rather than taking a vector and only
+                    // operating on the lower bits.ther than taking a vector and only
+                    // operating on the lower bits.
+
+                    assert(supportsAlignedSIMDLoads == false);
+                    assert(supportsUnalignedSIMDLoads == false);
+                    assert(supportsSIMDScalarLoads == false);
+
+                    const unsigned expectedSize = genTypeSize(containingNode->gtSIMDBaseType);
+                    const unsigned operandSize  = genTypeSize(node->TypeGet());
+
+                    supportsGeneralLoads = (operandSize == expectedSize);
```

Comment:
Perhaps (operandSize >= expectedSize) is enough, and do not need the following if-block?

Reply:
The following block is to be more explicit, to give a place for the explanation comment, and to ensure it is only done when we expect a  8/16-bit operand (which is all it should be valid for).

It will also provide a place for any fix ups if the check isnt the right one.

Reply:
But the current may lose the chance of containment with `Int64`->`Int16/Byte` or `Int16`->`Byte` casts.

Reply:
I think that's already the case, since we seem to generally have an explicit `CAST` for narrowing. I'll wait for feedback when people get back from holiday, as there may be a better way to approach this.

Reply:
What exactly are you trying to contain here? I would guess that's an IND node, in that case morph should have eliminated the cast. Example code?